controllers/kyma_controller.go

```diff
@@ -122,13 +122,19 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	return r.reconcile(ctx, kyma)
 }
 
-//nolint:cyclop
+//nolint:funlen,cyclop,gocognit
 func (r *KymaReconciler) reconcile(ctx context.Context, kyma *v1beta2.Kyma) (ctrl.Result, error) {
-	if r.SyncKymaEnabled(kyma) {
+	if r.SyncKymaEnabled(kyma) { //nolint:nestif
 		var err error
 		remoteClient := remote.NewClientWithConfig(r.Client, r.KcpRestConfig)
 		if ctx, err = remote.InitializeSyncContext(ctx, kyma,
 			r.RemoteSyncNamespace, remoteClient, r.RemoteClientCache); err != nil {
+			if !kyma.DeletionTimestamp.IsZero() && apierrors.IsNotFound(err) {
```

Comment:
I am thinking about reducing the nested if-statement. Did you think somehow about switch case for this scenario?

Reply:
@medmes  How would I use a switch for this case?